libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -762,9 +763,19 @@ bool RCOutput::setup_group_DMA(pwm_group &group, uint32_t bitrate, uint32_t bit_
         group.pwm_started = false;
     }
 
-    // original prescaler calculation from betaflight. bi-dir dshot is incredibly sensitive to the bitrate
     const uint32_t target_frequency = bitrate * bit_width;
-    uint32_t prescaler = uint32_t(lrintf((float) group.pwm_drv->clock / (bitrate * bit_width) + 0.01f) - 1);
+    uint32_t prescaler = group.pwm_drv->clock / target_frequency;
+    if (is_dshot) {
+        // original prescaler calculation from betaflight. bi-dir dshot is incredibly sensitive to the bitrate
+        prescaler = uint32_t(lrintf((float) group.pwm_drv->clock / (bitrate * bit_width) + 0.01f) - 1);
+    } else {
+        // adjust frequency to give an allowed value given the clock.
+        const uint32_t clock_hz = group.pwm_drv->clock;
+        while ((clock_hz / prescaler) * prescaler != clock_hz && prescaler <= 0x8000) {
```

Comment:
I think this needs to be < clock_hz otherwise they may never match

Reply:
i just restored the old, well-tested, code that is known to work with LEDs and serial